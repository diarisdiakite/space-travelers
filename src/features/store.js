import { configureStore } from '@reduxjs/toolkit';
import missionsReducer from './missions/missionsSlice';
import usersReducer from './users/usersSlice';
import rocketsSlice from './rockets/rocketsSlice';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
    users: usersReducer,
    rockets: rocketsSlice,
  },
});

export default store;
