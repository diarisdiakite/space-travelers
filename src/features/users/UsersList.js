import { Link, useParams } from 'react-router-dom';
import { useSelector } from 'react-redux';
import { selectAllUsers } from './usersSlice';
import { selectMissionsByUser } from '../missions/missionsSlice';

function UsersList() {
  const { userId } = useParams();
  const users = useSelector(selectAllUsers);

  const userMissions = useSelector((state) => selectMissionsByUser(state, Number(userId)));

  const content = users.map((user) => (
    <div key={user.id}>
      <h2 key={user.id}>
        <Link to={`/users/${user.id}`}>
          {user.name}
          {' '}
          (
          {userMissions.length}
          )
        </Link>
      </h2>
      <p key={`description-${user.id}`}>
        {user.description}
        {' '}
      </p>
    </div>
  ));

  return (
    <div className="main-container">
      {content}
    </div>
  );
}

export default UsersList;
