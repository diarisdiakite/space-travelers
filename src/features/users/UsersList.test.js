import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../store';
import UsersList from './UsersList';

describe('UsersList component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <UsersList />
          </MemoryRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
