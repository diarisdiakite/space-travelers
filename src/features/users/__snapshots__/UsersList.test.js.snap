// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList component renders correctly 1`] = `
<div
  className="main-container"
>
  <div>
    <h2>
      <a
        href="/users/1"
        onClick={[Function]}
      >
        My profile
         
        (
        0
        )
      </a>
    </h2>
    <p>
      User's description
       
    </p>
  </div>
</div>
`;
